### DISCUSSION TITLE
How do Qmd files work with blogging?

### DISCUSSION THREAD

**Discussion by AndreasThinks** [quarto-dev/quarto-cli#2624] [Q&A]
Seeking some advice! I've got a blog that was previously a combination of jupyter and rmarkdown notebooks (converted to html) hosted through Nikola that I want to migrate to quarto, and I'm trying to understand how Quarto converts Qmd files. 

I understand I can put both the ipynb files and the qmd files in the post folder of my blog, and then render it...but as they're all quite computationally intensive, will that then "re-render" them? I think adding the "freeze" variable will stop that happening, but do I then still use the qmd file, or should I instead put in the outputted html file?  A little confused.

**Comment by cderv** [comment]
A small note on how Quarto handles `.ipynb`: https://quarto.org/docs/computations/python.html#rendering
> Note that when rendering an .ipynb Quarto will not execute the cells within the notebook by default (the presumption being that you already executed them while editing the notebook).

So they won't be rerendered unless you ask for it.

Regarding [freeze](https://quarto.org/docs/projects/code-execution.html#freeze), you still use the .qmd file. it will allow you to not render by default when doing a project render. 

Did you try already putting you files in a Blog ?

**Reply by AndreasThinks** [reply]
so the issue I'm finding is the it keeps trying to "rerun" the Qmd file, even though I've already run it in it's entirety... 

I presume it's because i ran it in a different folder (where the dependencies are?)

**Reply by cderv** [reply]
> I presume it's because i ran it in a different folder (where the dependencies are?)

Can you share more on the structure ? I am not sure I am following.

Just to me more precise, as documented https://quarto.org/docs/projects/code-execution.html#freeze, freeze will not re-render  when you run the whole project `quarto render`. If you call `quarto render post.qmd` or `quarto render subfolder/` it will re-render (we call that project render vs incremental render). When using the feature, everything is in `_freeze` folder so be sure to keep it around.

**Reply by AndreasThinks** [reply]
so to make sure I've got this right... say I have a qmd notebook "project_notebook" in my "data" folder, along with a bunch of csv files etc. add my
freeze: true and then render the notebook (either in rstudio or via the cli). It outputs the html file

Now that the notebook is rendered, I'd like to move the qmd file to my "blog/posts" folder so I can use it as a post, but without re-rendering (and needing to see all the data sources etc). Looking at the freeze documentation, it looks like I should expect a _freeze directory to be created in the initial folder, that I'm assuming could be moved wiht it and prevent re-rendering, but I'm not seeing it... though I've just spotted there is a "project_notebook_cache" which I'm hoping will work!

**Reply by cderv** [reply]
Oh now I get it. You want to move output files around with all there resources ? 
I don't think `freeze` feature will support this. 

Why do you need to do that, why not have the qmd directly in the blog/posts directly ?  

>  it looks like I should expect a _freeze directory to be created in the initial folder

This will be created project wise, so in the root project folder. 

> though I've just spotted there is a "project_notebook_cache" which I'm hoping will work!

There are also caching mechanism possible which is a bit different, but could maybe be used better if you want to move your source around.
